pkg/controller/deployment/deployment_controller.go

```diff
@@ -102,36 +332,46 @@ func (d *DeploymentController) reconcileRollingUpdateDeployment(deployment exten
 	allRCs := append(oldRCs, newRC)
 
 	// Scale up, if we can.
-	scaledUp, err := d.reconcileNewRC(allRCs, newRC, deployment)
+	scaledUp, err := dc.reconcileNewRC(allRCs, newRC, deployment)
 	if err != nil {
 		return err
 	}
 	if scaledUp {
 		// Update DeploymentStatus
-		return d.updateDeploymentStatus(allRCs, newRC, deployment)
+		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 
 	// Scale down, if we can.
-	scaledDown, err := d.reconcileOldRCs(allRCs, oldRCs, newRC, deployment)
+	scaledDown, err := dc.reconcileOldRCs(allRCs, oldRCs, newRC, deployment)
 	if err != nil {
 		return err
 	}
 	if scaledDown {
 		// Update DeploymentStatus
-		return d.updateDeploymentStatus(allRCs, newRC, deployment)
+		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 	// TODO: raise an event, neither scaled up nor down.
```

Comment:
We should also update status if deployment.status.replicas != desiredReplicas or deployment.status.updatedReplicas != desiredReplicas.
That should fix the e2e failure that you are seeing.